### DISCUSSION TITLE
Neovim Error

### DISCUSSION THREAD

**Discussion by augustocdias** [topgrade-rs/topgrade#281] [Q&A]
Topgrade hangs forever when trying to update my Neovim plugins. After I press `Control+C` I get the following error:

```
Neovim failed:
   0: Command failed: `/opt/homebrew/bin/nvim -u /Users/augusto/.config/nvim/init.lua --headless -V1 -nS /var/folders/nq/9crflph97xngvp0kvpjy35400000gn/T/.tmpK5aXTg`

      Stderr:
      Error detected while processing BufWinEnter Autocommands for "*":
      E5108: Error executing lua Vim:E495: no autocommand file name to substitute for "<afile>"
      stack traceback:
        [C]: in function 'expand'
        [string ":lua"]:1: in main chunk
      Error detected while processing BufEnter Autocommands for "*":
      E5108: Error executing lua Vim:E495: no autocommand file name to substitute for "<afile>"
      stack traceback:
        [C]: in function 'expand'
        [string ":lua"]:1: in main chunk
      TreeSitter UpdatePacker
   1: `/opt/homebrew/bin/nvim` failed: signal: 2 (SIGINT)

Location:
   src/executor.rs:169
```
What could be causing this?

**Comment by s34m** [comment]
This looks more line a Neovim or plugin error then a topgrade error to me but idk

**Reply by augustocdias** [reply]
I agree... But shouldn't topgrade implement some kind of timeout? This hangs forever...

**Reply by s34m** [reply]
The problem with timeouts is to choose what a valid length is. On slow internet and low powered hardware it could easily be that even a vim update takes 5-10 minutes or even more, but if we choose like 15 minutes for example a timeout doesn't really makes sense any more.

**Comment by riotrah** [comment]
So I actually have a _similar_ issue. Topgrade overall works quite well but always fails to upgrade neovim - on mac m1 at least.

It appears to hang without any logging for many, many minutes (I usually switch apps by that point), and it has never succeeded. My internet and machine are extremely fast. Running `:PackerSync` manually within nvim runs within a minute or two, and generally without issue.

But when I `ctrl-c` - it shows a different error from OP, but fails the same way each time nonetheless. If I can remember to when I get back to my mac, I can post a sample.
